src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1244,8 +1253,10 @@ static int volume_prepare(struct processing_module *mod)
 	for (i = 0; i < cd->channels; i++) {
 		cd->volume[i] = cd->vol_min;
 		volume_set_chan(mod, i, cd->tvolume[i], false);
-		if (cd->volume[i] != cd->tvolume[i])
+		if (cd->volume[i] != cd->tvolume[i]) {
 			cd->ramp_finished = false;
+			cd->copy_needed = true;
```

Comment:
Actually, this file you set all cd->copy_needed  = true no matter cd->ramp_finished is true or false, it is confused to me. And for those settings behind cd->ramp_finished = false, I think it is not needed since you will set it at volume_ramp. Can you check about this?